qa-tests-backend/src/main/groovy/util/CollectorUtil.groovy

```diff
@@ -0,0 +1,105 @@
+        setExternalIps(orchestrator, DISABLED_VALUE)
+        waitForConfigToHaveState(orchestrator, DISABLED_VALUE, timeoutSeconds)
+    }
+
+    static deleteRuntimeConfig(Kubernetes orchestrator) {
+        orchestrator.deleteConfigMap(RUNTIME_CONFIG_MAP_NAME, Constants.STACKROX_NAMESPACE)
+        return true
+    }
+
+    private static parseJsonToProtobuf(String json) {
+        Collector.CollectorConfig.Builder builder = Collector.CollectorConfig.newBuilder()
+        JsonFormat.parser().merge(json, builder)
+        return builder.build()
+    }
+
+    private static introspectionQuery(String collectorAddress, String endpoint) {
+        String uri = "http://${collectorAddress}${endpoint}"
+        URL url = new URL(uri)
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection()
+
+        // this might be unneeded?
+        connection.setRequestMethod("GET")
+        connection.connect()
+
+        if (connection.responseCode != HttpURLConnection.HTTP_OK) {
+            throw new RuntimeException("IntrospectionQuery failed with ${connection.responseMessage}")
+        }
+        def jsonResponse = connection.inputStream.text
+        return parseJsonToProtobuf(jsonResponse)
+    }
+
+    private static waitForConfigToHaveState(Kubernetes orchestrator, String state, int timeout = 90, int port = 8080) {
+        def portForwards = orchestrator.createCollectorPortForwards(port)
+
+        log.info "Waiting for Collector Config propagation (${portForwards.size()} pods)"
+        int intervalSeconds = 1
+        int waitTime = 0
+        int nretry = (timeout / intervalSeconds).toInteger()
+        withRetry(nretry, intervalSeconds) {
+            // if a pod has the right config, remove it from the list
+            // we need to check
+            portForwards.removeAll {
+                def config = introspectionQuery("127.0.0.1:${it.getLocalPort()}", "/state/runtime-config")
+                def configTyped = (Collector.CollectorConfig) config
+                return configTyped.networking.externalIps.enabled.name() == state
+            }
+            waitTime += intervalSeconds
+            assert portForwards.isEmpty()
+        }
+
+        def success = portForwards.isEmpty()
+        if (success) {
+            log.info "Waited for ${waitTime} seconds for Collector runtime configuration to be updated"
+        } else {
+            log.info "Waiting for Collector runtime configuration timed out after ${timeout} seconds"
+        }
+
+        // if we timed out, some collectors have not updated
+        // the config, so return false
+        return success
+    }
+
+    static private setExternalIps(Kubernetes orchestrator, String state) {
+        String runtimeConfig = """|
+                                  |networking:
+                                  |  externalIps:
+                                  |    enabled: ${state}
+                                  |""".stripMargin()
```

Comment:
It would be a bit more extensible to use a yaml serialiser rather than dealing with strings directly

Reply:
I created this ticket https://issues.redhat.com/browse/ROX-29045